message.go

```diff
@@ -406,6 +406,15 @@ type Arena interface {
 	// arena is responsible for preserving the existing data in the
 	// returned byte slice.
 	Allocate(minsz Size, segs map[SegmentID]*Segment) (SegmentID, []byte, error)
+
+	// Return this arena to an internal sync.Pool of arenas that can be
+	// re-used, which can help reduce memory allocations.
```

Comment:
I'd kindof like to keep this a little more abstract, and find a way to  express this without reference to the exact implementation. In particular, you've defined this as a no-op on roSingleSegment and testArena, which seems fine but is inconsistent with the docs.

Reply:
Ah, that's a good thought.  Fixed.

Reply:
Hm, I worry the new version is a little misleading and suggests something more like msg.Reset(); the caller should not directly re-use the arena.

Reply:
Baah, you're right.  In fact, I got myself mixed up, as you can see.